### PR TITLE
add test functions in context_test.go

### PR DESCRIPTION
**Pull request by NickrenREN**
add test functions in context_test.go( pkg/api/context_test.go)

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/37188)

### CODE REVIEW

**Comment by dims** [comment]
@k8s-bot ok to test

**Comment by NickrenREN** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by NickrenREN** [comment]
@erictune   Jenkins kops AWS e2e build failed.  Details show  that   waiting for nodes timed out .   Not related to the test code .

**Comment by NickrenREN** [comment]
@erictune Jenkins kops AWS e2e build failed. Details show that kops configuration failed .
two checks failed .   Not related to the test code .

**Comment by NickrenREN** [comment]
@erictune  checks passed . please review

**Comment by NickrenREN** [comment]
@erictune   @dims  I add some test functions. And checks  passed .  Could one of you help with the review ?

File: pkg/api/context_test.go [hunk 1]
```diff
@@ -66,3 +68,67 @@ func TestValidNamespace(t *testing.T) {
 		t.Errorf("Expected the empty string")
 	}
 }
+
+//TestUserContext validates that a userinfo can be get/set on a context object
+func TestUserContext(t *testing.T) {
+	ctx := api.NewContext()
+	_, ok := api.UserFrom(ctx)
+	if ok {
+		t.Errorf("Should not be ok because there is no user.Info on the context")
```
Comment (caesarxuchao) [inline pkg/api/context_test.go]: Use `t.Fatalf` to stop further testing.

File: pkg/api/context_test.go [hunk 2]
```diff
@@ -66,3 +68,67 @@ func TestValidNamespace(t *testing.T) {
...
+			Groups: []string{"group1"},
+			Extra:  map[string][]string{"foo": {"bar"}},
+		},
+	)
+
+	result, ok := api.UserFrom(ctx)
+	if !ok {
+		t.Errorf("Error getting user info")
+	}
+	if result.GetName() != "bob" {
```
Comment (caesarxuchao) [inline pkg/api/context_test.go]: Can you test other fields as well?

**Comment by NickrenREN** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/api/context_test.go, line 77 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/37188#-KX3qdLK0bVrobz3riAT-r1-77:-KZZ9bvphLmH1bGRrrVU:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/e700acae7246114a31b9de3d809970a345be3d75/pkg/api/context_test.go#L77)):*
<details><summary><i>Previously, caesarxuchao (Chao Xu) wrote…</i></summary><blockquote>

Use `t.Fatalf` to stop further testing.
</blockquote></details>

Done.

---

*[pkg/api/context_test.go, line 93 at r1](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/37188#-KX3qdLK0bVrobz3riAT-r1-93:-KZZ9eJT-DGnFHozWjEN:b-896fix) ([raw file](https://github.com/kubernetes/kubernetes/blob/e700acae7246114a31b9de3d809970a345be3d75/pkg/api/context_test.go#L93)):*
<details><summary><i>Previously, caesarxuchao (Chao Xu) wrote…</i></summary><blockquote>

Can you test other fields as well?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.kubernetes.io:443/reviews/kubernetes/kubernetes/37188)*

**Comment by NickrenREN** [comment]
@caesarxuchao   Thanks a lot for your advice. Codes were modified.  Please review

**Comment by caesarxuchao** [comment]
Thanks. LGTM.

**Pull request merged**